### PR TITLE
enhance: Support MinIO TLS connection

### PR DESCRIPTION
**Pull request by yhmo**
issue: https://github.com/milvus-io/milvus/issues/30709
pr: #31292

### CODE REVIEW

**Comment by yanliang567** [comment]
manual pass for it reran for title updated

File: pkg/util/paramtable/service_param.go [hunk 1]
```diff
@@ -1094,15 +1095,23 @@ func (p *MinioConfig) Init(base *BaseTable) {
...
-		Key:          "minio.useSSL",
-		Version:      "2.0.0",
+		Key:          "minio.ssl.enabled",
+		Version:      "2.3.12",
 		DefaultValue: "false",
 		PanicIfEmpty: true,
 		Doc:          "Access to MinIO/S3 with SSL",
 		Export:       true,
 	}
 	p.UseSSL.Init(base.mgr)
```
Comment (congqixia) [inline pkg/util/paramtable/service_param.go:1105]: Add fallback key to be compatible with old instance

File: internal/datacoord/garbage_collector_test.go [hunk 2]
```diff
@@ -248,6 +249,14 @@ func Test_garbageCollector_scan(t *testing.T) {
 // initialize unit test sso env
 func initUtOSSEnv(bucket, root string, n int) (mcm *storage.MinioChunkManager, inserts []string, stats []string, delta []string, other []string, err error) {
 	paramtable.Init()
+
+	if Params.MinioCfg.UseSSL.GetAsBool() && len(Params.MinioCfg.SslCACert.GetValue()) > 0 {
+		err := os.Setenv("SSL_CERT_FILE", Params.MinioCfg.SslCACert.GetValue())
```
Comment (congqixia) [inline internal/datacoord/garbage_collector_test.go:254]: Why is this env needed? cannot find any reference or usage of it.

Comment (yhmo) [reply internal/datacoord/garbage_collector_test.go:254]: minio-go use this env to pass the certificate file path:
https://github.com/minio/minio-go/blob/e8ddcf0238962d766f44242b595511f4decd365c/transport.go#L73C22-L73C35

File: pkg/util/paramtable/service_param.go (hunk 1 above)
Comment (yhmo) [reply pkg/util/paramtable/service_param.go]: FallbackKeys: []string{"minio.useSSL"},

File: configs/milvus.yaml [hunk 3]
```diff
@@ -68,7 +68,9 @@ minio:
   port: 9000 # Port of MinIO/S3
   accessKeyID: minioadmin # accessKeyID of MinIO/S3
   secretAccessKey: minioadmin # MinIO/S3 encryption string
-  useSSL: false # Access to MinIO/S3 with SSL
+  ssl:
+    enabled: false # Access to MinIO/S3 with SSL
+    tlsCACert: /path/to/public.crt # path to your CACert file, ignore when it is empty
```
[the remark this answers is not in the document]
Comment (yhmo) [reply configs/milvus.yaml:73]: this style is following the etcd, kafka configurations

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by congqixia** [comment]
maybe we could change to cert file param to manual composing a transport later

**Pull request merged**